test/codeu/chat/TestRunner.java

```diff
@@ -32,6 +32,7 @@ public static void main(String[] args) {
          );
       for (final Failure failure : result.getFailures()) {
          System.out.println(failure.toString());
+         System.exit(-1);
```

Comment:
The test runner should print all the failures before exiting. Can you move this below `System.out.println(result.wasSuccessful());` and use `result.wasSuccessful()` to determine what the return value exit code should be?